src/components/views/room_settings/RelatedGroupSettings.js

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2017 New Vector Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// import Promise from 'bluebird';
+import React from 'react';
+import {MatrixEvent, MatrixClient} from 'matrix-js-sdk';
+// var ObjectUtils = require("../../../ObjectUtils");
+// var MatrixClientPeg = require('../../../MatrixClientPeg');
+import sdk from '../../../index';
+import { _t } from '../../../languageHandler';
+import Modal from '../../../Modal';
+
+const GROUP_ID_REGEX = /\+\S+\:\S+/;
+
+module.exports = React.createClass({
+    displayName: 'RelatedGroupSettings',
+
+    propTypes: {
+        roomId: React.PropTypes.string.isRequired,
+        canSetRelatedRooms: React.PropTypes.bool.isRequired,
+        relatedGroupsEvent: React.PropTypes.instanceOf(MatrixEvent),
+    },
+
+    contextTypes: {
+        matrixClient: React.PropTypes.instanceOf(MatrixClient),
+    },
+
+    getDefaultProps: function() {
+        return {
+            canSetRelatedRooms: false,
+        };
+    },
+
+    getInitialState: function() {
+        return {
+            newGroupsList: this.props.relatedGroupsEvent ?
+                (this.props.relatedGroupsEvent.getContent().groups || []) : [],
+            newGroupId: null,
+        };
+    },
+
+    saveSettings: function() {
+        return this.context.matrixClient.sendStateEvent(
+            this.props.roomId,
+            'm.room.related_groups',
+            {
+                groups: this.state.newGroupsList,
+            },
+            '',
+        );
+    },
+
+    validateGroupId: function(groupId) {
+        const localDomain = this.context.matrixClient.getDomain();
+        if (!GROUP_ID_REGEX.test(groupId) || !groupId.endsWith(localDomain)) {
```

Comment:
why can I only add a group on my own server as a related group?

Reply:
Apparently this is unnecessary so we can get rid of the check.